### PR TITLE
collect certs on disk: set owner, permissions and selinux label, skip revisioned folders

### PR DESCRIPTION
**Pull request by vrutkovs**
Tested in https://github.com/openshift/origin/pull/28523

### CODE REVIEW

**Comment by deads2k** [comment]
/approve

If you want to update for types like https://github.com/openshift/library-go/pull/1656 (or something similar) now is probably the right time.

File: pkg/certs/cert-inspection/certgraphanalysis/collector.go [hunk 1]
```diff
@@ -216,23 +244,73 @@ func MergePKILists(ctx context.Context, first, second *certgraphapi.PKIList) *ce
...
-		CertificateAuthorityBundles: *caBundlesList,
-		CertKeyPairs:                *certList,
+		CertificateAuthorityBundles: deduplicateCABundlesList(caBundleList),
+		CertKeyPairs:                deduplicateCertKeyPairList(certList),
+		OnDiskResourceData:          deduplicateOnDiskMetadata(onDiskResourceData),
 		InClusterResourceData:       inClusterData,
 	}
 }
+
+func stripPrefixFromPaths(certList *certgraphapi.CertKeyPairList, caBundlesList *certgraphapi.CertificateAuthorityBundleList, onDiskResourceData *certgraphapi.PerOnDiskResourceData, param pathUpdateParams, pathFuncs ...pathUpdateFn) {
```
Comment (deads2k) [inline pkg/certs/cert-inspection/certgraphanalysis/collector.go:280]: so this isn't prefixes, it's `mutatePathFns`?

File: pkg/certs/cert-inspection/certgraphanalysis/collector.go [hunk 2]
```diff
@@ -216,23 +244,73 @@ func MergePKILists(ctx context.Context, first, second *certgraphapi.PKIList) *ce
...
+
+	onDiskResourceData := &certgraphapi.PerOnDiskResourceData{
+		TLSArtifact: append(first.OnDiskResourceData.TLSArtifact, second.OnDiskResourceData.TLSArtifact...),
+	}
+
+	param := pathUpdateParams{
+		prefix:   prefix,
+		nodeList: nodeList,
+	}
+	stripPrefixFromPaths(certList, caBundleList, onDiskResourceData, param,
```
Comment (deads2k) [inline pkg/certs/cert-inspection/certgraphanalysis/collector.go:266]: Why did this move from when we collect data to when we merge it?  Doing it at collection adjusts how every layer in the stack sees the data. Versus doing it here where it means other locations in the code see the old names.

File: pkg/certs/cert-inspection/certgraphanalysis/deduplication.go [hunk 3]
```diff
@@ -183,3 +183,20 @@ func CombineCABundleOnDiskLocations(in *certgraphapi.CertificateAuthorityBundle,
...
 	return out
 }
+
+func deduplicateOnDiskMetadata(in *certgraphapi.PerOnDiskResourceData) certgraphapi.PerOnDiskResourceData {
+	ret := certgraphapi.PerOnDiskResourceData{}
+	for _, curr := range in.TLSArtifact {
+		found := false
+		for _, existing := range ret.TLSArtifact {
+			if existing.Path == curr.Path {
+				found = true
```
Comment (deads2k) [inline pkg/certs/cert-inspection/certgraphanalysis/deduplication.go:193]: break after?

**Comment by deads2k** [comment]
moving the strip to the merge bothers me: https://github.com/openshift/library-go/pull/1661#discussion_r1459602967

I'm not clear on why it should happen there instead of with our other mutating collection options.

Squash up to something more reasonable please.

File: pkg/certs/cert-inspection/certgraphanalysis/collector.go (hunk 2 above)
Comment (vrutkovs) [reply pkg/certs/cert-inspection/certgraphanalysis/collector.go:266]: That would require all collection layers to be aware of rootfsmount (doable) and require node list permissions. Its implementable, but that makes us dedup and filter twice - once on the node and once for incluster resources

Comment (deads2k) [reply pkg/certs/cert-inspection/certgraphanalysis/collector.go:266]: Let's push  it into the collection path.

**Pull request merged**